.github/workflows/test.yaml

```diff
@@ -0,0 +1,88 @@
+name: Tests stage - unit test for sniffs
+
+on:
+    pull_request:
+        branches: [ master, develop ]
+    push:
+        branches: [ master ]
+
+jobs:
+    tests:
+        name: Unit tests on PHP ${{ matrix.php-versions }} with PHPCS ${{ matrix.phpcs-versions }} and WPCS ${{ matrix.wpcs-versions }}
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                php-versions: [ '5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1' ]
+                phpcs-versions: [ '3.5.0', 'dev-master' ]
+                wpcs-versions: [ '2.3.0', 'dev-master', 'dev-develop']
+                exclude:
```

Comment:
I struggled with what to do here in the WPCS workflow. I [elected to exclude certain branches from the matrix in favor of using `include` statements](https://github.com/WordPress/WordPress-Coding-Standards/pull/1965/files#diff-cd0effce06f425599e96952f4d5f684641a683f46d9909e7c74a9f8b2f03ccd3R30). The number of `includes` that result was less than the number of `excludes` combinations that way. Not sure if that would also be true here as well.

Reply:
Yes, noticed that in the WPCS PR and it definitely looks a lot cleaner. I'll refactor this. For the first pass, I tried having the same output from the Travis here in GH actions so this was a quick and dirty way of achieving this 😄